cupy/random/_generator_api.pyx

```diff
@@ -195,6 +198,35 @@ class Generator:
         # omitted args.
         return (<object>y).astype(dtype, copy=False)
 
+    def standard_normal(self, size=None, dtype=numpy.float64, out=None):
+        """Returns an array of samples drawn from the standard normal distribution.
+
+        Args:
+            size (int or tuple of ints): The shape of the array. If ``None``, a
+                zero-dimensional array is generated.
+            dtype: Data type specifier.
+
+            out (cupy.ndarray, optional): If specified, values will be written
+                to this array
+
+        Returns:
+            cupy.ndarray: Samples drawn from the standard normal distribution.
+
+        .. seealso:: :func:`numpy.random.standard_normal`
+
+        """
+        cdef ndarray y
+
+        y = ndarray(size if size is not None else (), numpy.float64)
+        _launch_dist(self.bit_generator, standard_normal, y, ())
+        if out is not None:
+            out[...] = y
+            y = out
+        # we cast the array to a python object because
+        # cython cant call astype with the default values for
+        # omitted args.
+        return (<object>y).astype(dtype, copy=False)
```

Comment:
`out` is not returned when `dtype` does not match.  For consistency with other methods, I'm leaving it in this PR.

See #4903 for details.

Reply:
Thanks!

Reply:
Added a check in #4904, do we wait and fix this pr too?

Reply:
Thanks.  #4904 is merged.  Let's fix it in this PR.

Since other dtypes are rejected even if `out` is given, I suppose `astype` can be removed.

```
>>> rng = np.random.default_rng(0)
>>> rng.standard_normal((2, 3), out=np.empty((2, 3), np.int32), dtype=np.int32)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "_generator.pyx", line 1020, in numpy.random._generator.Generator.standard_normal
TypeError: Unsupported dtype dtype('int32') for standard_normal
```